cdap-ui/server/aggregator.js

```diff
@@ -43,59 +49,104 @@ Aggregator.prototype.log = function () {
 };
 
 /**
- * schedule polling
+ * Checks if the 'id' received from the client is already registered -- This
+ * check was added because for whatever reason 'Safari' was sending multiple
+ * requests to backend with same ids. As this is happens only once during
+ * the start of poll, it's safe to make this check. The assumption here is
+ * that the frontend is sending in unique ids within the websocket session.
+ *
+ * Upon check if it's not duplicate, we invoke doPoll that would make the
+ * first call and set the interval for the timeout.
+ */
+Aggregator.prototype.startPolling = function (resource) {
+  // WARN: This assumes that the browser side ids are unique for a websocket session.
+  // This check is needed for Safari.
+  if(this.polledResources[resource.id]) {
+    // console.log("Resource id " + resource.id + " already registered.");
+    return;
+  }
+  resource.interval = resource.interval || POLL_INTERVAL;
+  this.polledResources[resource.id] = resource;
+  _.bind(doPoll, this, resource)();
```

Comment:
Might be [performance](https://jsperf.com/bind-vs-jquery-proxy/68)
2x speed difference. Not a big deal, due to the rate at which we call them.